lib/rrdd_plugin.ml

```diff
@@ -147,15 +147,30 @@ module Reporter = struct
 			(module D : Debug.DEBUG)
 			?(neg_shift=0.5)
 			~uid
-			~protocol =
+			~protocol ~overdue_count =
 		let next_reading =
 			RRDD.Plugin.Local.register uid Rrd.Five_Seconds protocol
 		in
 		let wait_time = next_reading -. neg_shift in
 		let wait_time = if wait_time < 0.1 then wait_time+.5. else wait_time in
-		if wait_time > 0. then Thread.delay wait_time
-		else
-			D.debug "rrdd says next reading is overdue by %.1f seconds; not sleeping" (-.wait_time)
+		(* overdue count - 0 if there is no overdue; +1 if there is overdue *)
+		let updated_overdue = ref 0 in
```

Comment:
You can get rid of this ref by having the first branch of the following if statement contain

```
Thread.delay wait_time;
0
```

and have the second branch simply return `overdue_count + 1` rather than storing it in the ref.